presto-main/src/main/java/com/facebook/presto/operator/aggregation/DoubleSumAggregation.java

```diff
@@ -25,13 +25,29 @@
     private DoubleSumAggregation() {}
 
     @InputFunction
-    @IntermediateInputFunction
     public static void sum(NullableDoubleState state, @SqlType(StandardTypes.DOUBLE) double value)
     {
         state.setNull(false);
         state.setDouble(state.getDouble() + value);
     }
 
+    @CombineFunction
+    public static void combine(NullableDoubleState state, NullableDoubleState otherState)
+    {
+        if (state.isNull()) {
+            if (otherState.isNull()) {
+                return;
+            }
+            state.setNull(false);
```

Comment:
why not just remove the condition above and do `state.setNull(otherState.isNull)`? I think it is more readable and looks the same as similar code for double. Will it throw on calling `getDouble` for the nullable value?